dftimewolf/lib/collectors/virustotal.py

```diff
@@ -0,0 +1,215 @@
+
+import os
+import tempfile
+import urllib.parse
+import zipfile
+
+from typing import List
+from typing import Optional
+
+import vt
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class VTCollector(module.BaseModule):
+  """VirusTotal (VT) Collector.
+
+  Attributes:
+    hashes_list List[hashes_list]: List of hashes passed ot the module
+    vt_type: pcap or evtx depending on the file type requested
+
+  """
+
+  def __init__(
+      self,
+      state: DFTimewolfState,
+      name: Optional[str],
+      critical: bool = False) -> None:
+    """Initializes an VirusTotal (VT) collector.
+
+    Args:
+      state: recipe state.
+      name: The module's runtime name.
+      critical: True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(VTCollector, self).__init__(state, name=name, critical=critical)
+    self.hashes_list: List[str] = []
+    self.directory: Optional[str] = None
+    self.client: Optional[vt.Client] = None
+    self.vt_type = None
+
+  def Process(self) -> None:
+    """Process of the VirusTotal collector after setup"""
+
+    for vt_hash in list(self.hashes_list):
+      if not self._isHashKnownToVT(vt_hash):
+        self.logger.info(
+            f'Hash not found on VT removing element {vt_hash} from list')
+        self.hashes_list.remove(vt_hash)
+
+    self.logger.info(f'Found the following files on VT: {*self.hashes_list,}')
+
+    for vt_hash in self.hashes_list:
+      pcap_download_list = self._getDownloadLinks(vt_hash)
+
+    for download_link in pcap_download_list:
+      self.logger.info(urllib.parse.quote(download_link))
+      filename = f'{vt_hash}.{self.vt_type}'
+      file = open(os.path.join(self.directory, filename), "wb")
+
+      download = self.client.get(download_link)
+      if download.status == 200:
+        file_content = download.content.read()
+
+        if len(file_content) == 0:
+          continue
+
+        file.write(file_content)
+
+      else:
+        self.logger.info(f'File not found {urllib.parse.quote(download_link)}')
+
+      if self.vt_type == 'pcap':
+        container = containers.File(name=vt_hash, path=file.name)
+        self.state.StoreContainer(container)
+
+      if self.vt_type == 'evtx':
+        # Unzip the file so that plaso can go over EVTX part in the archive
+        extract_output_dir = f'{file.name}_extract'
+        if not os.path.isdir(extract_output_dir):
+          os.makedirs(extract_output_dir)
+
+        with zipfile.ZipFile(file.name) as archive:
+          archive.extractall(path=extract_output_dir)
+          self.logger.debug(
+              f'Downloaded file extracted to {extract_output_dir}')
+
+        container = containers.File(
+            name=vt_hash, path=os.path.abspath(extract_output_dir))
```

Comment:
You're storing a directory in a File container. Does this make sense? Would it be better to create a new `Directory` container type? I assume that a lot of consumers of `containers.File` will just call `open` on it's `.file` attribute, which might behave unpredictably if called on a directory.